apis/v1beta1/vspherefailuredomain_types.go

```diff
@@ -43,6 +44,19 @@ type VSphereFailureDomainSpec struct {
 
 	// Topology describes a given failure domain using vSphere constructs
 	Topology Topology `json:"topology"`
+
+	// VirtualMachineTemplate defines parameters for VM
+	// +optional
+	VMTemplate VirtualMachineTemplate `json:"vmTemplate"`
```

Comment:
Why do we need a separate struct here?

VirtualMachineTemplate could be very confusing.

Reply:
I thought it can be used in future for other fields that override machines. But if adding plain field seems better it is ok.